### PR TITLE
Init Pass on  MaxSrcResolution for Issue 8

### PR DESCRIPTION
**Pull request by bscott**
Working PR to discuss the defaults for Resolution vs using Dimension.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -69,10 +69,10 @@ type config struct {
 	MaxClients      int
 	TTL             int
 
-	MaxSrcDimension int
-
-	Quality         int
-	GZipCompression int
+	MaxSrcDimension  int
+	MaxSrcResolution int
+	Quality          int
```
Comment (DarthSim) [inline config.go]: Don't remove separating empty lines, please.

File: config.go [hunk 2]
```diff
@@ -111,7 +112,7 @@ func init() {
 	intEnvConfig(&conf.TTL, "IMGPROXY_TTL")
 
 	intEnvConfig(&conf.MaxSrcDimension, "IMGPROXY_MAX_SRC_DIMENSION")
-
+	intEnvConfig(&conf.MaxSrcResolution, "IMGPROXY_MAX_SRC_RESOLUTION")
```
Comment (DarthSim) [inline config.go]: Don't remove separating empty lines, please.

File: config.go [hunk 3]
```diff
@@ -81,15 +81,16 @@ type config struct {
...
-	Quality:         80,
-	GZipCompression: 5,
+	Bind:             ":8080",
+	ReadTimeout:      10,
+	WriteTimeout:     10,
+	DownloadTimeout:  5,
+	Concurrency:      runtime.NumCPU() * 2,
+	TTL:              3600,
+	MaxSrcDimension:  4096,
+	MaxSrcResolution: 65,
```
Comment (DarthSim) [inline config.go]: 65 megapixels is quite hight for default IMHO. Let's set it to 15.

**Comment by DarthSim** [comment]
And thanks for your contribution! :)

**Comment by homm** [comment]
> `MaxSrcDimension` doesn't make much sense

I don't know VIPS internals but for some libraries, this does sense, because too tall or too wide images (for example 1x15000000) can have some overhead in processing and this could be used as a vulnerability.

I suggest setting the max resolution to some suitable value, like 15M and set the max dimension to a high, but realistic value like 16K or 32K.

File: config.go (hunk 3 above)
Comment (homm) [reply config.go]: It used to be MaxSrcDimension* MaxSrcDimension, i.e. 16 megapixels.

**Comment by DarthSim** [comment]
Closed with https://github.com/DarthSim/imgproxy/commit/2940478ec9ef298129a6e6df1c58e688cd03c478

**Pull request closed without merging**